### PR TITLE
Justus Exercise_02.py

### PR DESCRIPTION
**Pull request by Saiki77**

### CODE REVIEW

File: Exercises/exercises_02.py [hunk 1]
```diff
@@ -40,3 +74,42 @@
...
+
+def categorize_strings(*args):
+    numbers = []
+    single_characters = []
+    
+    for arg in args:
+        # Checking if the string can be interpreted as a number (int, float, or hex)
+        try:
+            # Checking for hexadecimal format (starting with 0x or -0x)
+            if arg.startswith(("0x", "-0x")):
```
Comment (atrejut) [inline Exercises/exercises_02.py:89]: nice :D

File: Exercises/exercises_02.py [hunk 2]
```diff
@@ -25,11 +25,45 @@
...
+        if opcode == 99:
+            # Opcode 99: halting
+            break
+        elif opcode == 1:
+            # Opcode 1: addition
+            value1 = memory[memory[pos + 1]]
+            value2 = memory[memory[pos + 2]]
+            memory[memory[pos + 3]] = value1 + value2
+        elif opcode == 2:
+            # Opcode 2: multiplication
```
Comment (atrejut) [inline Exercises/exercises_02.py:42]: some duplicate code here between the two functional if-branches. You could try to think of a way to reduce the duplication

**Review by atrejut — CHANGES_REQUESTED** [review]
you're modifying the exercise template. Please make a copy in your own subfolder instead

File: Exercises/exercises_02.py [hunk 3]
```diff
@@ -25,11 +25,45 @@
...
+            value1 = memory[memory[pos + 1]]
+            value2 = memory[memory[pos + 2]]
+            memory[memory[pos + 3]] = value1 * value2
+        else:
+            raise ValueError(f"Unexpected opcode {opcode} at position {pos}")
```
Comment (FabianPerl93) [inline Exercises/exercises_02.py:47]: I really like your idea of implementing an error message, that gives back information about what went wrong.

**Review by FabianPerl93 — COMMENTED** [review]
I like your code and the test cases you used. In my opinion, some of the comments were a bit unnecessary, but that's just personal taste.

File: Exercises/exercises_02.py [hunk 4]
```diff
@@ -40,3 +74,42 @@
...
+def categorize_strings(*args):
+    numbers = []
+    single_characters = []
+    
+    for arg in args:
+        # Checking if the string can be interpreted as a number (int, float, or hex)
+        try:
+            # Checking for hexadecimal format (starting with 0x or -0x)
+            if arg.startswith(("0x", "-0x")):
+                # Converting hex to integer and append to numbers
```
Comment (FabianPerl93) [inline Exercises/exercises_02.py:90]: I like the idea of additionally checking for hexadecimal numbers.

**Review by atrejut — APPROVED** [review]
thanks for fixing the file issues

**Pull request merged**